File: back/src/main/java/com/back/nbe141team5/order/entity/Order.java
package com.back.nbe141team5.order.entity;

import com.back.nbe141team5.global.BaseEntity;
import jakarta.persistence.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "orders")
public class Order extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String email;

    private LocalDateTime orderDate;

    private String status;

    private Integer totalPrice;

    private LocalDateTime deliveryDate;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    private List<OrderItem> orderItems = new ArrayList<>();

    public Order() {
    }
}

File: front/types/order.ts
export type OrderStatus = "ORDERED" | "SHIPPED" | "DELIVERED";

export type OrderItem = {
  id: number;
  productId: number | null;
  productName: string;
  quantity: number;
};

export type Order = {
  id: number;
  email: string;
  address: string;
  postcode: string;
  orderDate: string;
  status: OrderStatus;
  totalPrice: number;
  deliveryDate: string | null;
  orderItems: OrderItem[];
};

export const ORDER_STATUS_LABEL: Record<OrderStatus, string> = {
  ORDERED: "주문 완료",
  SHIPPED: "배송 중",
  DELIVERED: "배송 완료",
};
